packages/core/src/index.ts

```diff
@@ -114,6 +114,7 @@ const fromInternal = (internalState: InternalState): State =>
 type Subscriber = (response: Array<Response>) => void;
 
 export const createConversation = (config: Config): ConversationHandler => {
+  let socket: WebSocket;
```

Comment:
Declare this as `WebSocket | undefined` because there is no guarantee it is defined at any point in the flow.